geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala

```diff
@@ -216,17 +223,25 @@ object QueryPlanner extends LazyLogging {
       // clear any configured hints so we don't process them again
       threadedHints.clear()
     }
+    // handle any params passed in through geoserver
+    QueryPlanner.handleGeoServerParams(query)
     // set transformations in the query
     QueryPlanner.setQueryTransforms(query, sft)
     // set return SFT in the query
     QueryPlanner.setReturnSft(query, sft)
-    // handle any params passed in through geoserver
-    QueryPlanner.handleGeoServerParams(query)
+  }
 
+  /**
+   * Return a new query with updated filters, ready to execute
+   *
+   * @param base query to update
+   * @param sft simple feature type to be queried
+   * @return new query with filters updated
+   */
+  def updateFilter(base: Query, sft: SimpleFeatureType): Query = {
+    val query = new Query(base)
     // add the bbox from the density query to the filter
-    // if configure has been called already, don't re-add - sometimes the bbox is split by
-    // IDL handling, causing us to not detect it as a duplicate
-    if (query.getHints.isDensityQuery && !query.getHints.isConfigured) {
+    if (query.getHints.isDensityQuery) {
```

Comment:
just checking if the comment from before about IDL handling is no longer relevant?